src/v/utils/delta_for.h

```diff
@@ -483,6 +483,11 @@ class deltafor_encoder
         return tmp;
     }
 
+    friend bool operator==(
+      const deltafor_encoder<TVal, DeltaStep, use_nttp_deltastep, delta_alg>&,
+      const deltafor_encoder<TVal, DeltaStep, use_nttp_deltastep, delta_alg>&)
+      = default;
```

Comment:
@Lazin: what do you think about this? Looking at the members the default kind of makes sense to me, but I might be missing things.

Reply:
I had some doubts if `mutable iobuf _data{};` would be included in the defaulted operator==, but I just checked, and it gets included in the comparison.

btw who is using this method?

Reply:
The new raft heartbeats which use the `deltafor_encoder`.